amgut/lib/data_access/sql_connection.py

```diff
@@ -53,6 +53,7 @@ def __init__(self):
         self._connection = None
         self._post_commit_funcs = []
         self._post_rollback_funcs = []
+        self.auto_rollback = False
```

Comment:
All the tests are modifying the database, but were designed in such a way as to either use other functions to reset as part of the testing of both functions or other ways of making sure the tests can be run multiple times.